src/System.Private.CoreLib/shared/System/Runtime/Loader/LibraryNameVariation.Windows.cs

```diff
@@ -0,0 +1,29 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#nullable enable
+using System.Collections.Generic;
+
+namespace System.Runtime.Loader
+{
+    internal partial struct LibraryNameVariation
+    {
+        private const string LibraryNameSuffix = ".dll";
+
+        internal static IEnumerable<LibraryNameVariation> DetermineLibraryNameVariations(string libName, bool isRelativePath)
+        {
+            // This is a copy of the logic in DetermineLibNameVariations in dllimport.cpp in CoreCLR
```

Comment:
This does not match what the unmanaged DetermineLibNameVariations does (both Windows and Unix versions). It would nice to fix that.

cc @jkoritzinsky

Reply:
Filed #24187. I can just wait it out until it's fixed.